src/Chainweb/BlockHeaderDB/PruneForks.hs

```diff
@@ -61,160 +99,162 @@ import Data.LogMessage
 -- This function doesn't guarantee to delete all blocks on forks. A small number
 -- of fork blocks may not get deleted.
 --
--- The function takes a callback that is invoked on each deleted block header.
--- The callback takes a parameter that indicates whether the block payload hash
--- is shared with any non-deleted block header. There is a small rate of false
--- positives of block payload hashes that are marked in use.
---
--- This doesn't update the the cut db or the payload db.
+-- The function takes callbacks that are invoked on each block header and
+-- paylaod hash. The callback takes a parameter that indicates whether the
+-- related block is pruned or not.
 --
 pruneForks
-    :: Logger logger
-    => logger
+    :: LogFunctionText
     -> BlockHeaderDb
     -> Natural
-        -- ^ The depth at which the roots are collected for marking block
-        -- headers that are kept. Any fork that was active that the given depth
-        -- is kept.
-        --
-        -- The depth is computed based on the entry of maximum rank. This entry
-        -- is not necessarily included in the overall consensus of the chainweb.
-        -- In particular the the higest ranking entry is not necessarily the
-        -- entry of largest POW weight.
-        --
-        -- Usually this number would be defined in terms of the chainweb
-        -- diameter and/or the difficulty adjustment window.
-    -> (BlockHeader -> Bool -> IO ())
+        -- ^ The depth at which deletion starts. Note, that the max rank isn't
+        -- necessarly included in the current best cut. So one, should choose a
+        -- depth for which one is confident that all forks are resolved.
+
+    -> (Bool -> BlockHeader -> IO ())
+        -- ^ Deletion call back. This hook is called /after/ the entry is
+        -- deleted from the database. It's main purpose is to delete any
```

Comment:
It looks like the `Bool` argument is whether it is deleted from the database *or not*, and it is called on both "pivots" and deleted items, meaning this comment is inaccurate as well as the one for the following callback?